File: SmartBrowser_java_eclipse/src/com/teuskim/sbrowser/TextButton.java
package com.teuskim.sbrowser;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.TextView;

public class TextButton extends TextView {
	
	private static int sNormal = 0xff000000;
	private static int sPressed = 0xff000000;

	public TextButton(Context context, AttributeSet attrs, int defStyle) {
		super(context, attrs, defStyle);
	}

	public TextButton(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public TextButton(Context context) {
		super(context);
	}
	
	public static void setTextColor(int normal, int pressed){
		sNormal = normal;
		sPressed = pressed;
	}

	@Override
	public boolean onTouchEvent(MotionEvent event) {
		switch(event.getAction()){
		case MotionEvent.ACTION_DOWN:
			setTextColor(sPressed);
			break;
		case MotionEvent.ACTION_UP:
		case MotionEvent.ACTION_CANCEL:
			setTextColor(sNormal);
			break;
		}
		return super.onTouchEvent(event);
	}
	
}

File: SmartBrowser_kotlin/app/src/main/java/com/teuskim/sbrowser/TextButton.kt
package com.teuskim.sbrowser

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.TextView

class TextButton : TextView {

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context) : super(context) {}

    override fun onTouchEvent(event: MotionEvent): Boolean {
        when (event.action) {
            MotionEvent.ACTION_DOWN -> setTextColor(sPressed)
            MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> setTextColor(sNormal)
        }
        return super.onTouchEvent(event)
    }

    companion object {

        private var sNormal = -0x1000000
        private var sPressed = -0x1000000

        fun setTextColor(normal: Int, pressed: Int) {
            sNormal = normal
            sPressed = pressed
        }
    }

}
